dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEventAnalyticsManager.java

```diff
@@ -88,7 +88,7 @@
     @Override
     public Grid getEvents( EventQueryParams params, Grid grid, int maxLimit )
     {
-        List<String> fixedCols = Lists.newArrayList( "psi", "ps", "executiondate", "longitude", "latitude", "ouname", "oucode" );
+        List<String> fixedCols = Lists.newArrayList( "psi", "tei", "pi", "ps", "executiondate", "longitude", "latitude", "ouname", "oucode" );
```

Comment:
Will this work for event programs (i.e. non-tracker programs) ?

Reply:
It wasnt 👎 
Updated the PR now, so that the PI and TEI columns are only included when the program has registration.